Fix previously broken handler functions and add tests

ryanlerch opened the pull request:
Got the list or rules to display again by returning a list.

Also then tried to add a new rule, and that failed too -- so started to try to fix that and failed. this is where i got to, and its probably all wrong.

Related: #609

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

ryanlerch committed:
Fix recently introduced API breakage

Don't manually cast to Pydantic API models, let FastAPI do its thing
with declared `response_model`.

Fixes: #609

Signed-off-by: Ryan Lerch <rlerch@redhat.com>
Signed-off-by: Aurélien Bompard <aurelien@bompard.org>
Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp commented on fmn/api/main.py:
```diff
@@ -109,9 +109,8 @@ async def get_user_rules(
-        await db_session.execute(Rule.select_related().filter(Rule.user.has(name=username)))
-    ).scalars()
+    db_result = await db_session.execute(Rule.select_related().filter(Rule.user.has(name=username)))
+    return [Rule.from_orm(rule) for rule in db_result.scalars()]
```
We shouldn't have to manually convert to API model objects, `response_model=` above should take care of it. And plain `Rule` is the ORM class, not the API model class.

nphilipp commented on fmn/api/main.py:
```diff
@@ -248,11 +247,9 @@ async def create_user_rule(
-    ).scalar_one()
+    db_result = await db_session.execute(Rule.select_related().filter(Rule.id == rule_db.id, Rule.user.has(name=username)))
+
+    return api_models.Rule.from_orm(db_result.scalar_one())
```
Same as above.

nphilipp reviewed (changes requested):
Also, let's do this in the context of #606.

nphilipp committed:
Clarify how get_user_info() is tested

Also, make the test function match the name of the function being
tested, don't list the mocker fixture when not using it and use symbolic
status codes rather than plain numbers.

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Test get_user_rules()

Related: #606, #609

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Test get_user_rule()

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Test and fix edit_user_rule()

The `.dict()` method of Pydantic models needs to be told to skip over
unset fields.

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Add test fixture mocking an API identity

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Test delete_user_rule()

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Test create_user_rule()

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Add fixtures creating test data in the DB

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Add and use fixtures for FASJSON data and mocking

Signed-off-by: Nils Philippsen <nils@redhat.com>
Signed-off-by: James Richardson <james02135@hotmail.com>

nphilipp committed:
Test get_user()

Related: #606

Signed-off-by: Nils Philippsen <nils@redhat.com>
Signed-off-by: James Richardon <james02135@hotmail.com>

nphilipp committed:
Make API results consistent

Previously, get_users() and get_user() wrapped their results in a
descriptive dictionary, but the other request handlers don’t do that.

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp committed:
Add and use API model for users

This is simplified for now, i.e. doesn’t return the rules because I need
to figure out how to do that properly with the async DB session.

Signed-off-by: Nils Philippsen <nils@redhat.com>

abompard committed:
Remove leftover debugging print

Signed-off-by: Aurélien Bompard <aurelien@bompard.org>

nphilipp committed:
Test get_applications()

Signed-off-by: Nils Philippsen <nils@redhat.com>

nphilipp reviewed (approved):
LGTM (but I would say that 😉)
